lib/publiccloud/google.pm

```diff
@@ -0,0 +1,83 @@
+# SUSE's openQA tests
+#
+# Copyright © 2018 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Helper class for Google Cloud Platform
+#
+# Maintainer: Clemens Famulla-Conrad <cfamullaconrad@suse.de>
+#             Jose Lausuch <jalausuch@suse.de>
+
+package publiccloud::google;
+use Mojo::Base 'publiccloud::provider';
+use testapi;
+use strict;
+use utils;
+
+has account             => undef;
+has project_id          => undef;
+has private_key_id      => undef;
+has private_key         => undef;
+has service_acount_name => undef;
+has client_id           => undef;
+
+sub init {
+    my ($self)           = @_;
+    my $credentials_file = 'google_credentials.json';
+    my $credentials      = "{" . $/
+      . '"type": "service_account", ' . $/
+      . '"project_id": "' . $self->project_id . '", ' . $/
+      . '"private_key_id": "' . $self->private_key_id . '", ' . $/
+      . '"private_key": "' . $self->private_key . '", ' . $/
+      . '"client_email": "' . $self->service_acount_name . '@' . $self->project_id . '.iam.gserviceaccount.com", ' . $/
+      . '"client_id": "' . $self->client_id . '", ' . $/
+      . '"auth_uri": "https://accounts.google.com/o/oauth2/auth", ' . $/
+      . '"token_uri": "https://oauth2.googleapis.com/token", ' . $/
+      . '"auth_provider_x509_cert_url": "https://www.googleapis.com/oauth2/v1/certs", ' . $/
+      . '"client_x509_cert_url": "https://www.googleapis.com/robot/v1/metadata/x509/' . $self->service_acount_name . '%40' . $self->project_id . '.iam.gserviceaccount.com"' . $/
+      . '}';
+    save_tmp_file($credentials_file, $credentials);
+    assert_script_run('source ~/.bashrc');
+    assert_script_run('ntpdate -s time.google.com');
```

Comment:
why we need this ?

Reply:
It's needed since the date in the VM differs too much from the Google Cloud Platform timezone. It basically avoids this error when authenticating: 
    There was a problem refreshing your current auth tokens: invalid_grant: Invalid JWT: Token must be a short-lived token (60 minutes) and in a reasonable timeframe. Check your iat and exp values and use a clock with skew to account for clock differences between systems.

Reply:
This is the time of our working VM not from the SUT.
But indeed it is interesting why we get a time diff here.